### ISSUE TITLE
Useless error message in Add Parametric Morphism

### ISSUE THREAD

**Issue by andrew-appel** [coq/coq#15899]
Coq version 8.15 and previous versions

I write, 

```
Add Parametric Morphism : Ebinary
  with signature eq ==> expr_equiv ==> expr_equiv ==> eq
  as Ebinary_mor.
```
and I get the error message,
`Syntax error: [term] expected after '=' (in [term]).`
in which the second equal sign after "signature eq" is highlighted.

It took an hour of debugging, experimention, and reading the manual before I found that the problem is that I did not say `Require Import Setoid.`

But in fact the manual does not say you need to `Require Import Setoid` in the [description](https://coq.github.io/doc/v8.15/refman/addendum/generalized-rewriting.html?highlight=add%20parametric%20morphism#coq:cmd.Add-Parametric-Morphism) of `Add Parametric Morphism`.  Only in the `Example` does `Require Import Setoid` happen to appear, without comment.

So, two bugs:
(1)  horrible and useless error message in Coq, when Add Parametric Morphism is used without Require Import Setoid
(2)  refman does not explicitly say that Require Import Setoid is needed before using Add Parametric Morphism

**Comment by Alizter** [comment]
FTR the refman does mention the need for `Require Import Setoid` for `Add Parametric Relation`. I would argue that the vernacular commands for setoid rewriting are really a legacy compatibility thing, since the implementation of setoid rewrite was different. Nowadays it just registers a typeclass instance in the background. In the refman we should therefore emphasize making instances of the correct classes rather than using this fragile vernacular interface. That also makes the need for `Require Import Setoid` more intuitive since the classes exist in those files.

**Comment by SkySkimmer** [comment]
Isn't that because the notation isn't there? Not especially about Parametric Morphism.
Maybe we could reserve the notation in init/notations, then Parametric Morphism would say `Library Coq.Setoids.Setoid has to be required first.`

**Comment by andrew-appel** [comment]
I disagree with the proposed solution, that is, "reserve ==> notation".
It would suffice that when the Add Parametric Morphism command is used, Setoid must already be required and imported, otherwise one gets an error message.

Perhaps it's a good idea anyway to reserve the ==> notation, so that the Add Parametric Morphism notation does not clash with some other notation; but it's not the simplest solution to the error message problem.
Therefore I suggest that, at the very least, you change the title of this issue report back to what it was.

**Comment by SkySkimmer** [comment]
The whole command cannot parse without reserving the notation, so putting restrictions on it has no effect.
